### ISSUE TITLE
Dismiss is not working when tap outside of popup window

### ISSUE THREAD

**Issue by cvezina** [zawadz88/MaterialPopupMenu#66]
The popup window is not dismissed when you tap outside. You can reproduce the problem using the sample provided.

minSdkVersion 21
targetSdkVersion 28

**Comment by DeweyReed** [comment]
I encounter the same problem when I test the app on a Lollipop emulator. Removing this line can help although it's not a good solution: https://github.com/zawadz88/MaterialPopupMenu/blob/master/material-popup-menu/src/main/java/androidx/appcompat/widget/MaterialRecyclerViewPopupWindow.kt#L239

**Comment by zawadz88** [comment]
@cvezina @DeweyReed thanks for reporting this.
I can reproduce this on Lollipop 5.0 emulator as well. I seems to be working on newer OS version so I'll need to investigate further.

**Comment by zawadz88** [comment]
It seems to be an issue on Lollipop only (5.0 and 5.1) - I've checked also API 19, 23, 25 & 28 and it's correct there on the emulator at least.

**Comment by zawadz88** [comment]
This seems to be an issue with clipping rounded backgrounds #47 .
I can see 2 solutions at the moment:

1. Use `clipToOutline` on Marshmallow+ only i.e. change:
```kotlin
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            dropDownList.clipToOutline = true
            // Move the background from popup to RecyclerView for clipToOutline to take effect.
            dropDownList.background = background
            popup.setBackgroundDrawable(null)
        }
```
to:
```kotlin
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            dropDownList.clipToOutline = true
            // Move the background from popup to RecyclerView for clipToOutline to take effect.
            dropDownList.background = background
            popup.setBackgroundDrawable(null)
        }
```
2. Set `popup.setBackgroundDrawable(null)` on Marshmallow only i.e.
```kotlin
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            dropDownList.clipToOutline = true
            // Move the background from popup to RecyclerView for clipToOutline to take effect.
            dropDownList.background = background
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                popup.setBackgroundDrawable(null)
            }
        }
```

1. Means rounded corners will work only on M+ while 2. means we'll add overdraw on Lollipop devices.
Before:
![Screenshot_1575648176](https://user-images.githubusercontent.com/3322260/70337086-b1ed8000-184a-11ea-9b36-8f71bb524961.png)
After:
![Screenshot_1575648288](https://user-images.githubusercontent.com/3322260/70337089-b1ed8000-184a-11ea-9670-52f3a3adfd7a.png)

@Tunous do you have any thoughts on that?

**Comment by Tunous** [comment]
Option 2 appears to be a better solution in this case. I’m not sure whether removing it is entirely safe but I also don’t remember if there were any other issues which this line was intended to resolve. Most likely these were on other Android version (if ever).

**Comment by zawadz88** [comment]
Fixed in 4.0.1.
Thanks all for contributing!
